tour_model_eval/confirmed_trips_eval_bins_clusters.py

```diff
@@ -20,54 +22,206 @@
+
+    return homo_score, comp_score, v_score
+
+
+# - sp2en=True: change Spanish to English
+# - cvt_pur_mo=True: convert purposes and replaced mode
+# - cutoff=True: choose to analyze bins above cutoff
+# - cutoff=None: analyze all bins
+# Note: for sp2en and cvt_pur_mo, set either one to be True as needed. cvt_pur_mo will change language first
+def v_measure_clusters(all_users,radius,sp2en=None,cvt_pur_mo=None):
+    homo_score = []
+    comp_score = []
+    v_score = []
+    for i in range(len(all_users)):
+        user = all_users[i]
+        filter_trips,sim = filter_data(user,radius)
+
+        # filter out users that haven't enough trips (at least 10) to analyze
+        if len(filter_trips) < 10:
+            homo_score.append(NaN)
+            comp_score.append(NaN)
+            v_score.append(NaN)
+            continue
+        sim.bin_data()
+        sim.delete_bins()
+        bin_trips = sim.newdata
+        bins = sim.bins
+
+        if len(bin_trips) < 10:
+            homo_score.append(NaN)
+            comp_score.append(NaN)
+            v_score.append(NaN)
+            continue
+
+        # clustering the data only based on sil score (min_cluster = 0) instead of bins number (len(bins))
+        feat = featurization.featurization(bin_trips)
+        min = 0
+        max = int(math.ceil(1.5 * len(bins)))
+        feat.cluster(min_clusters=min, max_clusters=max)
+        cluster_trips = feat.data
+        cluster_user_input_df = pd.DataFrame(data=[i["data"]["user_input"] for i in cluster_trips])
+        if sp2en:
+            # change language
+            cluster_user_input_df = cluster_user_input_df.replace(span_eng_dict)
+            cluster_user_input_ls = cluster_user_input_df.values.tolist()
+        elif cvt_pur_mo:
+            # change language first
+            cluster_user_input_df = cluster_user_input_df.replace(span_eng_dict)
+            # convert purpose
+            cluster_user_input_df = cluster_user_input_df.replace(map_pur_dict)
+            # convert mode
+            for a in range(len(cluster_user_input_df)):
+                if cluster_user_input_df.iloc[a]["replaced_mode"] == "same_mode":
+                    cluster_user_input_df.iloc[a]["replaced_mode"] = cluster_user_input_df.iloc[a]['mode_confirm']
+        # turn cluster_trips to list without any changes
+        cluster_user_input_ls = cluster_user_input_df.values.tolist()
+        # drop duplicate user_input
+        no_dup_df = cluster_user_input_df.drop_duplicates()
+        # turn non-duplicate user_input into list
+        no_dup_list = no_dup_df.values.tolist()
+        # collect labels_true based on user_input
+        labels_true = []
+        for trip in cluster_user_input_ls:
+            if trip in no_dup_list:
+                labels_true.append(no_dup_list.index(trip))
+        labels_pred = feat.labels
+
+        # compare the points in cluster_trips and those in feat.points, return nothing if two data frames are the same
```

Comment:
can you fix the comment here? Right now if the frames are the same, we just go on to the score calculation